routers/web/repo/issue_view.go

```diff
@@ -1006,3 +1017,24 @@ func prepareIssueViewContent(ctx *context.Context, issue *issues_model.Issue) {
 		return
 	}
 }
+
+func getPullViewSquashMergeCommits(ctx *context.Context, issue *issues_model.Issue) (string, error) {
+	pull := issue.PullRequest
+	pull.Issue = issue
+
+	baseCommit := GetMergedBaseCommitID(ctx, issue)
+
+	compareInfo, err := pull_service.GetCompareInfo(ctx, ctx.Repo.Repository, ctx.Repo.Repository, ctx.Repo.GitRepo, baseCommit, pull.GetGitHeadRefName(), false, false)
```

Comment:
GetCompareInfo is slow and it should have been called before.

It should reuse the previous result.